src/pyodide/internal/setupPackages.ts

```diff
@@ -31,26 +31,32 @@ export const STDLIB_PACKAGES: string[] = Object.values(LOCKFILE.packages)
 // `folder/file.txt` -> `["folder", "file.txt"]
 export type FilePath = string[];
 
+function createTarFs() {
+  return {
+    children: new Map(),
+    mode: 0o777,
+    type: '5',
+    modtime: 0,
+    size: 0,
+    path: '',
+    name: '',
+    parts: [],
+    reader: null,
+  };
+}
+
 /**
  * SitePackagesDir keeps track of the virtualized view of the site-packages
  * directory generated for each worker.
  */
 class SitePackagesDir {
   public rootInfo: TarFSInfo;
+  public dynlibTarFs: TarFSInfo;
   public soFiles: FilePath[];
   public loadedRequirements: Set<string>;
   constructor() {
-    this.rootInfo = {
-      children: new Map(),
-      mode: 0o777,
-      type: '5',
-      modtime: 0,
-      size: 0,
-      path: '',
-      name: '',
-      parts: [],
-      reader: null,
-    };
+    this.rootInfo = createTarFs();
+    this.dynlibTarFs = createTarFs();
```

Comment:
Would be possible to do:
```suggestion
    this.dynlibTarFs = structuredClone(this.rootInfo);
```
and keep the inline definition of `rootInfo`. Not sure that it's better.